src/backend/api/meals.js

```diff
@@ -1,15 +1,306 @@
 const express = require("express");
+const { limit, sum} = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+// Parse URL-encoded bodies (as sent by HTML forms)
+router.use(express.urlencoded({ extended: true }));
+// Parse JSON bodies (as sent by API clients)
+router.use(express.json());
+
+
+
+// router.get("/", async (request, response) => {
+//   try {
+//     const meals = await knex("meal").select("*");
+//     response.json(meals);
+//   } catch (error) {
+//     throw error;
+//   }
+// });
+
+router.get("/:id", async (request, response) => {
+  try {  
+    const meals = await knex("meal");
+    const inputNumber = Number(request.params.id);
+    const maxIdOfMeal =Math.max(meals.map(meal => Number(meal.id)));
+    if (isNaN(inputNumber)) {
+      response.send("not a number")
+    } 
+    else if (inputNumber > maxIdOfMeal) {
+      response.send(`the largest id is : ${maxIdOfMeal}`)
+    }
+    else {
+      const specificMeal = meals.filter((meal) => meal.id == inputNumber);
+      if (specificMeal.length > 0) {
+        response.json(specificMeal);
+      } else {
+        response.send("no such meal,pls check the id")
+      }
+    }
+  } catch (error) {
+    throw error;
+  }
+});
+
+// method 1 of post
+router.post("/", async (request, response) => {
```

Comment:
This will not work as you expect it. method 1 of post and method 2 of post have an identical route: '/' so they really cannot be differentiated. When you make a post request to the /meals route node is not able to know which of the two you want to use!

You should delete the method1 and only keep method 2 cause it seems method 1 is only inserting some mock data while the 2 uses the parameters passed

Reply:
Yes, Thanks for your advice. I will always use the parameters way.